app/feature_flags.go

```diff
@@ -21,7 +20,9 @@ func (s *Server) setupFeatureFlags() {
 	splitConfigured := splitKey != ""
 	syncFeatureFlags := splitConfigured && s.IsLeader()
 
-	s.configStore.PersistFeatures(splitConfigured)
+	if splitConfigured {
+		s.configStore.SetReadOnlyFF(false)
+	}
```

Comment:
```suggestion
	s.configStore.SetReadOnlyFF(!splitConfigured)
```
Otherwise this relies on an not intuitive default of true.